ffi/tests/firewood/merkle_compatibility_test.go

```diff
@@ -0,0 +1,442 @@
+package tests
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"path/filepath"
+	"slices"
+	"testing"
+
+	firewood "github.com/ava-labs/firewood-go/ffi"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/database/memdb"
+	"github.com/ava-labs/avalanchego/x/merkledb"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	keySize      = 4
+	valSize      = 4
+	maxBatchSize = 5
+	maxNumKeys   = 10_000
+)
+
+const (
+	dbGet byte = iota
+	dbUpdate
+	dbBatch
+	checkDBHash
+	createProposalOnProposal
+	createProposalOnDB
+	proposalGet
+	commitProposal
+	maxStep
+)
+
+var stepMap = map[byte]string{
+	dbGet:                    "dbGet",
+	dbUpdate:                 "dbUpdate",
+	dbBatch:                  "dbBatch",
+	checkDBHash:              "checkDBHash",
+	createProposalOnProposal: "createProposalOnProposal",
+	createProposalOnDB:       "createProposalOnDB",
+	proposalGet:              "proposalGet",
+	commitProposal:           "commitProposal",
+}
+
+func newTestDatabase(t *testing.T) *firewood.Database {
```

Comment:
nit: naming: `newFirewoodTestDatabase` or `newFwTestDb`

Important because we have two different databases here and I can't easily tell which this is for.